### PR TITLE
Parse C comments and mark deprecated field as Obsolete.

### PR DESCRIPTION
**Pull request by sdcb**

### CODE REVIEW

File: FFmpeg.AutoGen.CppSharpUnsafeGenerator/Definitions/StructureField.cs [hunk 1]
```diff
@@ -5,5 +5,7 @@ internal class StructureField : ICanGenerateXmlDoc
         public string Name { get; set; }
         public TypeDefinition FieldType { get; set; }
         public string Content { get; set; }
+        public bool IsDeprecated { get; set; }
```
Comment (Ruslan-B) [inline FFmpeg.AutoGen.CppSharpUnsafeGenerator/Definitions/StructureField.cs:8]: Must be named IsObsolete as it was already done in other structure.

File: FFmpeg.AutoGen.CppSharpUnsafeGenerator/Definitions/StructureField.cs [hunk 2]
```diff
@@ -5,5 +5,7 @@ internal class StructureField : ICanGenerateXmlDoc
...
+        public string DeprecatedMessage { get; set; }
```
Comment (Ruslan-B) [inline FFmpeg.AutoGen.CppSharpUnsafeGenerator/Definitions/StructureField.cs:9]: ObsoleteMessage

File: FFmpeg.AutoGen.CppSharpUnsafeGenerator/Processors/StructureProcessor.cs [hunk 3]
```diff
@@ -83,12 +83,21 @@ private void MakeDefinition(Class @class, string name)
                         continue;
                     }
 
+                    BlockCommandComment deprecatedComment = field.Comment?.FullComment.Blocks
```
Comment (Ruslan-B) [inline FFmpeg.AutoGen.CppSharpUnsafeGenerator/Processors/StructureProcessor.cs:86]: Copy paste, besides, please follow repository coding style

**Review by Ruslan-B — COMMENTED** [review]
After some analysis of the PR I think it should be done differently as deprecation/obsoletion done on different definitions as well. I'll try to work with you a bit to make it more generic.

**Comment by Ruslan-B** [comment]
Please make a contributor in your fork. So I can help you to pull it through.

File: FFmpeg.AutoGen.CppSharpUnsafeGenerator/Definitions/StructureField.cs (hunk 1 above)
Comment (sdcb) [reply FFmpeg.AutoGen.CppSharpUnsafeGenerator/Definitions/StructureField.cs]: Done.

File: FFmpeg.AutoGen.CppSharpUnsafeGenerator/Definitions/StructureField.cs (hunk 2 above)
Comment (sdcb) [reply FFmpeg.AutoGen.CppSharpUnsafeGenerator/Definitions/StructureField.cs]: Done.

File: FFmpeg.AutoGen.CppSharpUnsafeGenerator/Processors/StructureProcessor.cs (hunk 3 above)
Comment (sdcb) [reply FFmpeg.AutoGen.CppSharpUnsafeGenerator/Processors/StructureProcessor.cs]: Code follow as FunctionProcessor.cs, done.

**Comment by Ruslan-B** [comment]
Even though it is an improvement, I still see some communication issues here. Any way thank you for your work, based on what you discovered I created more generic version of deprecation/obsoletion.

**Pull request closed without merging**